docs/08-models-and-migrations.md

```diff
@@ -174,43 +177,72 @@ Let’s create a `PostFactory` in the `database/factories` folder:
 // 'database/factories/PostFactory.php'
 <?php
 
+namespace JohnDoe\BlogPackage\Database\Factories;
+
 use JohnDoe\BlogPackage\Models\Post;
-use Faker\Generator as Faker;
+use Illuminate\Database\Eloquent\Factories\Factory;
 
-$factory->define(Post::class, function (Faker $faker) {
-  return [
-    //
-  ];
-});
-```
+class PostFactory extends Factory
+{
+    /**
+     * The name of the factory's corresponding model.
+     *
+     * @var string
+     */
+    protected $model = Post::class;
 
-### Loading the Model Factory
+    /**
+     * Define the model's default state.
+     *
+     * @return array
+     */
+    public function definition()
+    {
+        return [
+            //
+        ];
+    }
+}
 
-To make use of the newly created model factory, you need to register them in the package's Service Provider. To add them to the Service Provider, point the `Illuminate\Database\Eloquent\Factory` class to the directory containing the model factories:
+```
 
-```php
-// 'src/BlogPackageServiceProvider.php'
+As with the `src` folder, for our package users to be able to use our model factories we'll need to register the `database/factories` folder within a namespace in our `composer.json` file:
 
-public function boot()
+```json
 {
-    // Register the model factories
-    $this->app->make('Illuminate\Database\Eloquent\Factory')
-        ->load(__DIR__.'/../database/factories');
-
-    // ...
+  ...,
+  "autoload": {
+    "psr-4": {
+      "JohnDoe\\BlogPackage\\": "src",
+      "JohnDoe\\BlogPackage\\Database\\Factories": "database/factories"
+    }
+  },
+  ...
 }
 ```
 
-If you only use model factories in your tests, you don't have to register them in the service provider. To access them in your tests, use the `withFactories()` method provided by Orchestra Testbench ([learn more](https://orchestraplatform.readme.io/docs/testbench#section-using-model-factories)) in the `setUp()` method of the parent `TestCase` class:
+After setting it up, don't forget to run `composer dump-autoload`.
 
-```php
-// 'tests/TestCase.php'
+### Configuring our Model factory
 
-protected function setUp()
-{
-    parent::setUp();
+Running our tests again leads to the following error:
+
+```
+Error: Class 'Database\Factories\JohnDoe\BlogPackage\Models\PostFactory' not found
+```
 
-    $this->withFactories(__DIR__.'/../database/factories');
+This is because Laravel is trying to resolve the Model class for our `PostFactory` assuming the default namespaces of a usual project (as of version 8.x, `App` or `App\Models`).
+To be able to instantiate the right Model from our package with the `Post::factory()` method, we need to add the following method to our `Post` Model:
+
+```php
```

Comment:
I would suggest to remove the doc block for clarity.

Reply:
Done!